File: benchmarks/fibonacci/fibonacci.py
# fibonacci_benchmark.py
# Python implementation of Fibonacci sequence benchmark

# Configuration
N1 = 10
N2 = 20
N3 = 30
N4 = 35

# Markers matching Lamina version
HEADER_MARKER = 123456789
FOOTER_MARKER = 987654321

def fibonacci_iterative(n):
    """Compute nth Fibonacci number iteratively."""
    if n == 0:
        return 0
    if n == 1:
        return 1

    a = 0
    b = 1
    for i in range(2, n + 1):
        temp = a + b
        a = b
        b = temp
    return b

def main():
    # Print header marker
    print(HEADER_MARKER)

    # Compute and print fibonacci numbers
    print(fibonacci_iterative(N1))
    print(fibonacci_iterative(N2))
    print(fibonacci_iterative(N3))
    print(fibonacci_iterative(N4))

    # Print footer marker
    print(FOOTER_MARKER)

if __name__ == "__main__":
    main()

File: benchmarks/fibonacci/fibonacci.js
// fibonacci_benchmark.js
// JavaScript implementation of Fibonacci sequence benchmark

// Configuration
const N1 = 10;
const N2 = 20;
const N3 = 30;
const N4 = 35;

// Markers matching Lamina version
const HEADER_MARKER = 123456789;
const FOOTER_MARKER = 987654321;

function fibonacci_iterative(n) {
    if (n === 0) return 0;
    if (n === 1) return 1;

    let a = 0, b = 1;
    for (let i = 2; i <= n; i++) {
        const temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

function main() {
    // Print header marker
    console.log(HEADER_MARKER);

    // Compute and print fibonacci numbers
    console.log(fibonacci_iterative(N1));
    console.log(fibonacci_iterative(N2));
    console.log(fibonacci_iterative(N3));
    console.log(fibonacci_iterative(N4));

    // Print footer marker
    console.log(FOOTER_MARKER);
}

main();
